src/librustc_codegen_llvm/back/lto.rs

```diff
@@ -261,7 +261,7 @@ fn fat_lto(cgcx: &CodegenContext,
         .max()
         .expect("must be codegen'ing at least one module");
     let module = modules.remove(costliest_module);
-    let mut serialized_bitcode = Vec::new();
+    let mut serialized_bitcode = Vec::with_capacity(serialized_modules.len());
```

Comment:
FWIW, this feels a little odd since we're later extending the serialized_modules vec in this function -- I think it definitely wants an explanation comment

Reply:
Actually it's `serialized_modules` that's later extended; `serialized_bitcode` is handled by a `push` loop.

Reply:
Right yeah -- so I think it's not "obvious" that this is the correct length so a note here as to why that is or perhaps using `.reserve` later on instead might work better.

Reply:
Ah, I didn't notice that; yeah, that's definitely commentworthy ^^.